crates/zombie_actor/src/ghost_test_harness.rs

```diff
@@ -290,18 +298,57 @@ macro_rules! wait1_for_callback {
 
         let is_match = regex.is_match(actual.as_str());
 
-        if is_match {
-            assert!(is_match);
-        } else {
-            assert_eq!($re, actual.as_str());
+        if $should_abort {
+            if is_match {
+                assert!(is_match);
+            } else {
+                assert_eq!($re, actual.as_str());
+            }
+        }
+        is_match
+    }};
+}
+
+#[allow(unused_macros)]
+#[macro_export]
+macro_rules! wait1_for_repeatable_callback {
```

Comment:
It took me quite a bit headbanging to understand how to use this function.  It's very cool, but I would have been helped by a comment above the macro definition explaining how build the request_fn and how it's return type works for passing the state along.  Alternatively the comment could go in the test.  I ended up having to triangulate the test (which is just a binary state) and the usage for websockets, which keeps changing the state.

Reply:
I totally understand that. I documented the macro and will provide simpler versions that build on top of it in subsequent PRs.